coaddAnalysis.py

```diff
@@ -0,0 +1,234 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+from astropy.visualization import (ZScaleInterval, SqrtStretch, ImageNormalize)
+
+import lsst.geom as geom
+from lsst.pipe.base import Struct
+
+__all__ = ("findPatches", "plotTractOutline", "mosaicCoadd")
+
+
+def findPatches(butler, skymap, verbose=True):
+    # Instead of using a manually defined patch list to figure out which patches
+    # have warps/coadds in them, go through each patch and try opening its coadd.
+    # If that succeeds, add the patch ID to our happy patch list.
```

Comment:
Please add a full function docstring. I like the verbose option.

Shouldn't gen3 "know" the patches that exist in a given skymap, and would it be easy to add a gen3 logic branch here? (OK to skip this for now if you can't work it out in like 15 min)

... and here we encounter the first case of me having written something that does this at least as well but not yet having pushed it anywhere useful since I keep bumping DM-25991 😬 Would you be willing to incorporate the following into your function so it's more readily useful for template lookup?

```
def getPatchConstituents(repo, printConstituents=False):
    """Learn which patches have data in them, and what dataIds (i.e., processed exposures) comprise each patch
    """
    dataIdSkymap = {'filter': 'g', 'tract': 0, 'patch': '0,0'}  # placeholder
    butler = dafPersist.Butler(repo)
    skymap = butler.get('deepCoadd_skyMap', dataId=dataIdSkymap)
    tractInfo = skymap[0]  # implicitly assumes there is just 1 tract
    everyPatchList = [str(patch.getIndex()[0]) + ',' + str(patch.getIndex()[1]) for patch in tractInfo]
    dataPatchList = []
    constituentList = []
    constituentCountList = []
    for patch in everyPatchList:
        try:
            dataIdPatch = {'filter': 'g', 'tract': 0, 'patch': patch}
            coadd_test = butler.get('deepCoadd', dataId=dataIdPatch)
            constituent = coadd_test.getInfo().getCoaddInputs().ccds['visit']
            constituentCount = len(constituent)
            if printConstituents:
                print(patch, constituentCount)#, constituent)
        except:
            #print('No data in this patch', patch)
            continue
        else:
            #print('This patch has a coadd', patch)
            dataPatchList.append(patch)
            constituentList.append(constituent)
            constituentCountList.append(constituentCount)
    return dataPatchList, constituentList, constituentCountList
```